woocommerce/woocommerce-ios
# Order Creation: Load stored products (real data) in Add Product screen

Part of: #5408

## Description

In Order Creation, the Add Product screen now displays a real list of products (loaded from Core Data). Some notes/caveats:

* For now we're only loading non-variable products. (Variable products will be implemented later.)
* The product list only shows published and private products. This matches core/web behavior, where draft and pending products can't be added to an order.
* For now we're only showing a placeholder product image. (Product images will be added in another PR.)
* This only loads products that already exist in storage. Since you can create a new order without having products already in Core Data, we'll need to sync products from remote. Because of the number of changes required for that sync, I'll add it in a separate PR.

## Changes

* Add Product screen:
   * Adds `AddProductViewModel` with results controller to load products from storage.
   * Updates the `AddProduct` view to display the list of products.
   * Shows an empty state view if the list of products is empty.
* New Order screen updated to use the `AddProduct` view model.
* Adds unit tests for the new view model.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Products tab to make sure your store's products are loaded in storage.
4. Go to the Orders tab and tap the + button. (If Simple Payments is also enabled, tap "Create order" in the bottom sheet that appears.)
5. On the New Order screen, tap the "Add product" button.
6. On the Add Product screen, confirm you see a complete list of published/private, non-variable products from your store.

Empty state:

To test the empty state screen, select a store with no products (or where products have not yet been loaded in the app) and repeat steps 4-6 to view the Add Product screen without products.

## Screenshots

Product list|Empty state
-|-
![Simulator Screen Shot - iPhone 13 Pro - 2021-12-02 at 14 44 32](https://user-images.githubusercontent.com/8658164/144448865-b8d14ee6-268a-4d37-8005-3e48fdb7d029.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-12-03 at 12 13 27](https://user-images.githubusercontent.com/8658164/144600749-1508fee5-dbd6-4966-8e2f-11bdf15a195d.png)

## Submitter Checklist

Update release notes:
